android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/ResourceExtension.kt

```diff
@@ -52,6 +55,11 @@ import timber.log.Timber
 
 private val fhirR4JsonParser = FhirContext.forR4Cached().getCustomJsonParser()
 
+fun DomainResource.findContained(id: String) = this.contained.first {
+  val searchableId = IdType(id)
+  it.idElement.idPart == searchableId.idPart
+}
+
 fun Base?.valueToString(): String {
   return when {
     this == null -> return ""
```

Comment:
This can be reduced to use elvis operator where it is used default to empty string.

```suggestion
//Example
theBase.valueToString() ?: ""
```